search-service/src/main/kotlin/com/egm/stellio/search/service/EntityTypeService.kt

```diff
@@ -0,0 +1,103 @@
+
+@Component
+class EntityTypeService(
+    private val databaseClient: DatabaseClient
+) {
+
+    suspend fun getEntityTypeList(contexts: List<String>): EntityTypeList {
+        val entityTypes = databaseClient.sql(
+            """
+            SELECT DISTINCT(types) FROM entity_payload
+            ORDER BY types
+            """.trimIndent()
+        ).allToMappedList { rowToTypes(it) }.flatten()
+
+        return EntityTypeList(typeList = compactTerms(entityTypes, contexts))
+    }
+
+    suspend fun getEntityTypes(contexts: List<String>): List<EntityType> {
+        val result = databaseClient.sql(
+            """
+            SELECT types, attribute_name
+            FROM entity_payload
+            JOIN temporal_entity_attribute ON entity_payload.entity_id = temporal_entity_attribute.entity_id
+            ORDER BY types
+            """.trimIndent()
+        ).allToMappedList { rowToEntityType(it) }.flatten().groupBy({ it.first }, { it.second }).toList()
+
+        return result.map {
+            EntityType(
+                id = toUri(it.first),
+                typeName = compactTerm(it.first, contexts),
+                attributeNames = compactTerms(it.second, contexts).toSet().sorted()
+            )
+        }
+    }
+
+    suspend fun getEntityTypeInfoByType(
+        type: String,
+        contexts: List<String>
+    ): Either<APIException, EntityTypeInfo> {
+        val expandedType = expandJsonLdTerm(type, contexts)
+        val result = databaseClient.sql(
+            """
+            WITH entities AS (
+                SELECT entity_id 
+                FROM entity_payload 
+                WHERE :type = any (types)
+                ORDER BY entity_id
+            )    
+            SELECT attribute_name, attribute_type, count(distinct(entity_id)) as count_entity
+            FROM temporal_entity_attribute
+            WHERE entity_id IN (SELECT entity_id FROM entities)
+            GROUP BY attribute_name, attribute_type
+            """.trimIndent()
+        )
+            .bind("type", expandedType)
+            .allToMappedList { it }
+
+        if (result.isEmpty())
+            return ResourceNotFoundException(typeNotFoundMessage(type)).left()
+
+        return EntityTypeInfo(
+            id = toUri(expandedType),
+            typeName = compactTerm(expandedType, contexts),
+            entityCount = (result.first()["count_entity"] as Long).toInt(),
+            attributeDetails = result.map {
+                AttributeInfo(
+                    id = toUri(it["attribute_name"]),
+                    attributeName = compactTerm(it["attribute_name"] as ExpandedTerm, contexts),
+                    attributeTypes = listOf(AttributeType.forKey(it["attribute_type"] as String))
+                )
+            }
+        ).right()
+    }
+
+    private fun rowToTypes(row: Map<String, Any>): List<ExpandedTerm> =
+        toList<ExpandedTerm>(row["types"]).filter { !AuthContextModel.IAM_TYPES.plus("Entity").contains(it) }
```

Comment:
this is no longer needed:
- the `Entity` label was a neo4j thing to identify nodes holding entities
- users, clients and groups are no longer entities (look at `EntityAccessRight` and `SubjectReferential`)